str/associatr/get_cis_numpy_files.py

```diff
@@ -0,0 +1,248 @@
+        ) / (len(gene_pheno))
+        # Use the inverse normal cumulative distribution function (quantile function) to transform percentiles to normal distribution values
+        gene_pheno.loc[:, 'gene_inverse_normal'] = norm.ppf(
+            gene_pheno.loc[:, 'gene_percentile']
+        )
+        gene_pheno = gene_pheno[['sample_id', 'gene_inverse_normal']]
+
+        gene_pheno_cov = gene_pheno.merge(covariates, on='sample_id', how='inner')
+
+        # filter for samples that were assigned a CPG ID; unassigned samples after demultiplexing will not have a CPG ID
+        gene_pheno_cov = gene_pheno_cov[
+            gene_pheno_cov['sample_id'].str.startswith('CPG')
+        ]
+
+        # add in cohort as a covariate
+        cohort_cpg_id = adata.obs[['cpg_id', 'cohort']]
+        mapping = {'TOB': 1, 'BioHEART': 2}
+        cohort_cpg_id['cohort'] = cohort_cpg_id['cohort'].map(mapping)
+        cohort_cpg_id.rename(columns={'cpg_id': 'sample_id'}, inplace=True)
+        cohort_cpg_id = cohort_cpg_id.drop_duplicates()
+
+        gene_pheno_cov = gene_pheno_cov.merge(
+            cohort_cpg_id, on='sample_id', how='inner'
+        )
+
+        gene_pheno_cov['sample_id'] = gene_pheno_cov['sample_id'].str[
+            3:
+        ]  # remove CPG prefix because associatr expects id to be numeric
+
+        gene_pheno_cov['sample_id'] = gene_pheno_cov['sample_id'].astype(float)
+
+        gene_pheno_cov = gene_pheno_cov.to_numpy()
+        with hl.hadoop_open(
+            output_path(
+                f'pheno_cov_numpy/{version}/{cell_type}/{chromosome}/{gene}_pheno_cov.npy'
+            ),
+            'wb',
+        ) as f:
+            np.save(f, gene_pheno_cov)
+
+
+@click.option('--input-h5ad-dir', help='GCS directory to the input AnnData objects')
+@click.option(
+    '--input-pseudobulk-dir', help='GCS directory to the input pseudobulk CSV files'
+)
+@click.option('--input-cov-dir', help='GCS directory to the input covariate CSV files')
+@click.option('--chromosomes', help=' eg chr22, comma separated if multiple')
+@click.option('--cell-types', help='cell type, comma separated if multiple')
+@click.option('--cis-window', help='cis window size in bp')
+@click.option('--version', default='v1', help='version of the output files')
+@click.option('--job-cpu', default=2, help='Number of CPUs to use for the job')
+@click.option('--job-memory', default='standard', help='Memory to use for the job')
+@click.option('--job-storage', default='5G', help='Storage to use for the job')
+@click.option(
+    '--max-parallel-jobs',
+    type=int,
+    default=500,
+    help=('To avoid exceeding Google Cloud quotas, set this concurrency as a limit.'),
+)
+@click.option(
+    '--min-pct',
+    default=1,
+    help='filter out genes that are expressed in fewer than XX% of cells',
+)
+@click.option(
+    '--remove-samples-file',
+    default=None,
+    help='GCS path to the file containing the list of samples to remove',
+)
+@click.command()
```

Comment:
it might be useful to shift to a config-oriented way of passing some arguments, like we did [here](https://github.com/populationgenomics/saige-tenk10k/pull/78) - instead of CLI arguments that have to be passed all over, you can just pop them in a config file, then pull them out at the point they're needed